modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/stored/DeleteStoredTemplateRequest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script.mustache.stored;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class DeleteStoredTemplateRequest extends AcknowledgedRequest<
+        DeleteStoredTemplateRequest> {
+
+    private String id;
+
+    DeleteStoredTemplateRequest() {
+    }
+
+    public DeleteStoredTemplateRequest(String id) {
+        this.id = id;
+    }
+
+    public DeleteStoredTemplateRequest id(String id) {
+        this.id = id;
+        return this;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+
+        if (id == null || id.isEmpty()) {
+            validationException = addValidationError("must specify id for stored search template",
+                    validationException);
+        } else
+            if (id.contains("#")) {
+                validationException = addValidationError(
+                        "id cannot contain '#' for stored search template", validationException);
+            }
```

Comment:
We don't have support for `_all` for either deleting stored scripts or stored templates. I can add a validation check and warn the user about it if you think it is worth doing.